cirq-core/cirq/optimizers/clifford_decomposition.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility methods to decompose Clifford gates into circuits."""
+
+from typing import List, TYPE_CHECKING
+import functools
+
+import numpy as np
+from cirq import ops, protocols, qis, sim
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def _X(
+    q: int,
+    args: sim.ActOnCliffordTableauArgs,
+    operations: List[ops.Operation],
+    qubits: List['cirq.Qid'],
+):
+    protocols.act_on(ops.X, args, qubits=[qubits[q]], allow_decompose=False)
+    operations.append(ops.X(qubits[q]))
+
+
+def _Z(
+    q: int,
+    args: sim.ActOnCliffordTableauArgs,
+    operations: List[ops.Operation],
+    qubits: List['cirq.Qid'],
+):
+    protocols.act_on(ops.Z, args, qubits=[qubits[q]], allow_decompose=False)
+    operations.append(ops.Z(qubits[q]))
+
+
+def _Sdg(
+    q: int,
+    args: sim.ActOnCliffordTableauArgs,
+    operations: List[ops.Operation],
+    qubits: List['cirq.Qid'],
+):
+    # Apply the tableau with S^\{dagger}
+    protocols.act_on(ops.S ** -1, args, qubits=[qubits[q]], allow_decompose=False)
+    operations.append(ops.S(qubits[q]))
+
+
+def _H(
+    q: int,
+    args: sim.ActOnCliffordTableauArgs,
+    operations: List[ops.Operation],
+    qubits: List['cirq.Qid'],
+):
+    protocols.act_on(ops.H, args, qubits=[qubits[q]], allow_decompose=False)
+    operations.append(ops.H(qubits[q]))
+
+
+def _CNOT(
+    q1: int,
+    q2: int,
+    args: sim.ActOnCliffordTableauArgs,
+    operations: List[ops.Operation],
+    qubits: List['cirq.Qid'],
+):
+    protocols.act_on(ops.CNOT, args, qubits=[qubits[q1], qubits[q2]], allow_decompose=False)
+    operations.append(ops.CNOT(qubits[q1], qubits[q2]))
+
+
+def _SWAP(
+    q1: int,
+    q2: int,
+    args: sim.ActOnCliffordTableauArgs,
+    operations: List[ops.Operation],
+    qubits: List['cirq.Qid'],
+):
+    protocols.act_on(ops.SWAP, args, qubits=[qubits[q1], qubits[q2]], allow_decompose=False)
+    operations.append(ops.SWAP(qubits[q1], qubits[q2]))
+
+
+def decompose_clifford_tableau_to_operations(
+    qubits: List['cirq.Qid'], clifford_tableau: qis.CliffordTableau
+) -> List[ops.Operation]:
+    """Decompose an n-qubit Clifford Tableau into a list of one/two qubit operations.
+
```

Comment:
Can you link out the paper in the docstring so if people look for this on the website it gets picked up ?

Reply:
Sure, added.